lms/services/vitalsource.py

```diff
@@ -0,0 +1,60 @@
+import oauthlib
+from oauthlib.oauth1 import SIGNATURE_HMAC_SHA1, SIGNATURE_TYPE_BODY
+
+
+class VitalSourceService:
+    def __init__(self, _context, request):
+        self._oauth_key = request.registry.settings["vitalsource_launch_key"]
+        self._oauth_secret = request.registry.settings["vitalsource_launch_secret"]
+
+    def get_launch_params(self, book_id, cfi, lti_user):
+        """
+        Launch the VitalSource book viewer using an LTI launch.
+
+        See https://developer.vitalsource.com/hc/en-us/articles/215612237-POST-LTI-Create-a-Bookshelf-Launch
+        """
+
+        launch_url = f"https://bc.vitalsource.com/books/{book_id}"
+        book_location = "/cfi" + cfi
+
+        launch_params = _launch_params(
+            user_id=lti_user.user_id,
+            roles=lti_user.roles,
+            # Set a dummy `context_id`. This needs to be replaced with the real
+            # course in future.
+            context_id="testcourse",
```

Comment:
According to the linked documentation above, the course information is optional but useful for analytics purposes:

> Context information will also allow VitalSource to report accurately and be able to distinguish between courses. Without the context information, VitalSource will see all launches as the same and without context, and therefore cannot provide any analytics about course activity.